### PR TITLE
docs: trace tests to requirements P+3

### LINKED ISSUE #18: `P+3`: Ensure that the new and updated tests are traced to requirements
Relevant test cases (existing tests and updated/new tests related to the refactored code) are traced to
requirements.

### PR DESCRIPTION
**Pull request by lskare**
add a section on how the implemented tests relate and trace back to the original requirements of the patch. The section shows that the 3 implemented tests fulfill all 5 initial requirements

Resolves #18

### CODE REVIEW

File: Fundsoft/report.md [hunk 1]
```diff
@@ -100,6 +100,20 @@ Scope (functionality and code affected).
 
 Optional (point 3): trace tests to requirements.
 
+* test_update_transitive()
+    * This test adds the dependency `requests` and later updates the sub-dependency `chardet`. Asserts that the sub-dependency is still not in the lock file and that the sub-dependency has been updated to the newest version from the virtual repository. Also asserts that the dependency `requests` has not been updated as this was not requested.
```
Comment (Stenmarken) [inline Fundsoft/report.md]: The test checks that the dependency is not in the pyproject.toml file. All sub-dependencies will still be in the lock file.

File: Fundsoft/report.md [hunk 2]
```diff
@@ -100,6 +100,20 @@ Scope (functionality and code affected).
...
+    * This test adds the dependency `requests` and later updates the sub-dependency `chardet`. Asserts that the sub-dependency is still not in the lock file and that the sub-dependency has been updated to the newest version from the virtual repository. Also asserts that the dependency `requests` has not been updated as this was not requested. 
+    * Fulfills requirements **ID 1** and **ID 5**
+
+* test_update_transitive_nonexistant_dependencies()
+    * This test adds the dependency `requests` and later attempts to update the non-existant sub-dependency `numpy`. This results in a "ProjectError" which is checked with two assertions. Thus throwing an error that the sub-dependency doesn't exist in the lock file.
+    * Fulfills requirements **ID 2**
+
+* test_update_transitive_non_transitive_dependencies()
+    * This test adds two dependencies `requests` and `pytz` and later updates both of these along with the sub-dependency `chardet`. It asserts that the sub-dependency is still not in the lock file and asserts that the sub-dependency and the two dependencies are both updated to newer versions from the virtual repository even when utilizing the flag "--allow_transitive".
```
Comment (Stenmarken) [inline Fundsoft/report.md]: Again, the sub-dependency should not be in the pyproject.toml file. We expect it to be in the lock file regardless of whether we update it or not.

**Review by Stenmarken — COMMENTED** [review]
Good job! I just had two comments, otherwise I thought everything was good!

**Comment by lskare** [comment]
Thanks for the feedback! I've now corrected this typo and written pyproject.toml file instead!

**Review by Stenmarken — APPROVED** [review]
Good job!

**Pull request merged**